pkg/aws/services/tagging_test.go

```diff
@@ -0,0 +1,421 @@
+package services
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestContainsTags(t *testing.T) {
+	tests := []struct {
+		contains   bool
+		findTags   Tags
+		sourceTags Tags
+		testName   string
+	}{
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			testName: "exact match with one tag",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "exact match with multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "match one tag from source's multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+				"Key3": aws.String("Value3"),
```

Comment:
just to mix this up can you please reverse the ordering here? So key3, key2, key1, just to make sure we aren't relying on ordering in our logic.

Reply:
Done